core/CronArchive.php

```diff
@@ -908,36 +907,30 @@ public function invalidateRecentDate($dateStr, $idSite)
         }
 
         $isYesterday = $dateStr === 'yesterday';
-        if ($isYesterday) {
-            // Skip invalidation for yesterday if archiving for yesterday was already started after midnight in site's timezone
-            $invalidationsInProgress = $this->model->getInvalidationsInProgress($idSite);
-            $today = Date::factoryInTimezone('today', $timezone);
-
-            foreach ($invalidationsInProgress as $invalidation) {
-                if (
-                    $invalidation['period'] == Day::PERIOD_ID
-                    && $date->toString() === $invalidation['date1']
-                    && Date::factory($invalidation['ts_started'], $timezone)->getTimestamp() >= $today->getTimestamp()
-                ) {
-                    $this->logger->debug("  " . ucfirst($dateStr) . " archive already in process for idSite = $idSite, skipping invalidation...");
-                    return;
-                }
-            }
-        }
```

Comment:
This check was incorrect here. On the one side it missed to check the done flag, causing any running archive to prevent a new validation. On the other side a running invalidation prevented any segment from being invalidated.
Therefor the check was moved to `invalidateWithSegments` method, where it is done for each segment separately.